src/Smart.Design.Razor/TagHelpers/Pagination/SmartPaginationHtmlGenerator.cs

```diff
@@ -0,0 +1,168 @@
+        pageLink.AddCssClass($"{(index == _settings.PageNumber ? "c-pagination__item--active" : string.Empty)}");
+
+        var anchorPageLink = new TagBuilder("a");
+        anchorPageLink.AddCssClass("c-pagination__link");
+        anchorPageLink.Attributes["href"] = GetQueryString(index);
+        anchorPageLink.InnerHtml.Append(index.ToString());
+
+        pageLink.InnerHtml.AppendHtml(anchorPageLink);
+        return pageLink;
+    }
+
+    private async Task<IHtmlContent> GetChevronLinkAsync(Image chevron)
+    {
+        var paginationItem = MakePaginationItem();
+        var chevronLink = new TagBuilder("a");
+        chevronLink.AddCssClass("c-button c-button--borderless c-button--icon");
+        chevronLink.Attributes["href"] = GetQueryString(chevron == Image.ChevronLeft ? _currentPageNumber - 1 : _currentPageNumber + 1);
+
+        if (chevron == Image.ChevronLeft && _currentPageNumber <= 1 || chevron == Image.ChevronRight && _currentPageNumber >= _settings.TotalPages)
+        {
+            chevronLink.Attributes["disabled"] = "disabled";
+        }
+
+        var span = new TagBuilder("span");
+        span.AddCssClass("c-button__content");
+        span.InnerHtml.AppendHtml(await _iconHtmlGenerator.GenerateIconAsync(chevron));
+
+        chevronLink.InnerHtml.AppendHtml(span);
+        paginationItem.InnerHtml.AppendHtml(chevronLink);
+
+        return paginationItem;
+    }
+
+    private TagBuilder MakePaginationItem()
+    {
+        var paginationItem = new TagBuilder("li");
+        paginationItem.AddCssClass("c-pagination__item");
+        return paginationItem;
+    }
+
+    private string GetQueryString(int pageNumber)
+    {
+        return $"{_settings.Path!.TrimEnd('/')}?{_settings.PageNumberParameterName}={pageNumber}&{_settings.QueryString}";
+    }
+
+    private void PrepareData()
+    {
+        _settings.Path ??= string.Empty;
+        SetCurrentPageInPossibleInterval();
+    }
+
+    private void SetCurrentPageInPossibleInterval()
+    {
+        _currentPageNumber = _settings.PageNumber < 1
+            ? Math.Min(_settings.PageNumber, 1)
+            : Math.Min(_settings.PageNumber, _settings.TotalPages);
+    }
+
+    private void ComputeNumberOfLinks()
+    {
+        // This makes sure to have the same number of links around the current page link.
+        _numberOfLinks = _settings.NumberOfLinks % 2 == 0
+            ? _settings.NumberOfLinks + 1
+            : _settings.NumberOfLinks;
+
+        _numberOfLinksAroundCurrentPageNumberLink = _numberOfLinks / 2;
+    }
+
+    private void ComputeEndIndex()
+    {
+        _endIndex = _settings.PageNumber <= _numberOfLinksAroundCurrentPageNumberLink
+            ? _numberOfLinks
+            : _currentPageNumber + _numberOfLinksAroundCurrentPageNumberLink;
+
+        _endIndex = Math.Min(_endIndex, _settings.TotalPages);
+    }
+
+    private void ComputeStartIndex()
+    {
+        _startIndex = Math.Max(_endIndex - _numberOfLinks + 1, 1);
```

Comment:
I guess it's intended but the statement should resolve to:
```suggestion
        _startIndex = Math.Max((_endIndex - _numberOfLinks) + 1, 1);
```
and not
```suggestion
        _startIndex = Math.Max(_endIndex - (_numberOfLinks + 1), 1);
```
So it might be clearer to read with the parenthis